pkg/config/refresher.go

```diff
@@ -63,7 +66,7 @@ func (r *refresher) refreshPeriodically(name string) {
 			err := r.fetchFunc()
 			if err != nil {
 				log.Error("can not pull configs", zap.Error(err))
-				r.intervalDone <- true
+				r.intervalDone <- struct{}{}
```

Comment:
Looks like this line may block forever since the `internalDone` is non-buffered channel. Maybe?
```suggestion
				r.stop()
```